### ISSUE TITLE
Character escaping issue

### ISSUE THREAD

**Issue by mariusz-schimke** [vfrz/DotNetGraph#13]
It would be wonderful if the solution automatically escaped strings used as labels and the title. I mean quote characters, line breaks, and others that I forgot to mention.

What I noticed so far is:
- diagram title is not quoted automatically when it contains spaces, and this breaks the structure of the compiled diagram,
- special characters in labels, and probably in the title as well, are not escaped (I did not check quoting when required).

I didn't familiarize with the DOT syntax enough, but from my experience with the library:
- line breaks have to be manually replaced with `\n` (some suggest using `<br />`, but it didn't work for me),
- quote characters have to be escaped with a backslash `"` -> `\"`,
- same case with backslashes: `\` -> `\\`
- I also escaped labels and title with `HttpUtility.HtmlEncode(label)` to handle special HTML characters like `<`, `>`, national characters (they also caused issues), etc.

Only then did the online visualizer http://www.webgraphviz.com/ stop reporting errors and warnings.

By the way, I love your library, it is so helpful! Cheers!

**Comment by mariusz-schimke** [comment]
Oh, it seems that HTML is interpreted in DOT syntax. Maybe differentiating between a `DotLabelAttribute` (with implicit HTML escaping) and a `DotHtmlLabelAttribute` (without HTML escaping) would be fine?

**Comment by vfrz** [comment]
Thanks for your bug report. I will take a look at the issue ASAP.

**Comment by vfrz** [comment]
I've published a draft PR #15 . For now it fixes backslash and double quote escaping in identifier and label.

I've two questions before continuing development: 
- What do you mean by replacing line breaks with `\n` ? Do you mean replacing windows newline (`\r\n`) with `\n` ?
- I don't know much about HTML in Dot and I havn't been able to make it work within webgraphviz compiler. Can you give me an example?

Thanks in advance.

**Comment by mariusz-schimke** [comment]
Thanks for the fixes! I posted an [answer](https://github.com/vfrz/DotNetGraph/pull/15/files#r415315170) to the first question in the PR.

If it comes to HTML in DOT it is a longer story, and I don't know the language well enough yet to give you authoritative suggestions, I'm sorry. I just noticed that http://www.webgraphviz.com/ has problems when a text label (as opposed to a HTML label) contains national characters for example. The `HttpUtility.HtmlEncode` method replaces them with their Unicode identifiers, but I'm not sure if it can be safely used here for all use cases.
